core/unit_test/incremental/Test16_ParallelScan.hpp

```diff
@@ -26,11 +26,45 @@ namespace Test {
 using value_type = double;
 const int N      = 10;
 
+template <typename ExecSpace>
+struct TrivialScanFunctor {
+  Kokkos::View<value_type *, ExecSpace> d_data;
+
+  KOKKOS_FUNCTION
+  void operator()(const int i, value_type &update_value,
+                  const bool final) const {
+    const value_type val_i = d_data(i);
+    if (final) d_data(i) = update_value;
+    update_value += val_i;
+  }
+};
+
+template <typename ExecSpace>
+struct NonTrivialScanFunctor {
+  Kokkos::View<value_type *, ExecSpace> d_data;
+
+  KOKKOS_FUNCTION
+  void operator()(const int i, value_type &update_value,
+                  const bool final) const {
+    const value_type val_i = d_data(i);
+    if (final) d_data(i) = update_value;
+    update_value += val_i;
+  }
+
+  NonTrivialScanFunctor()                              = default;
+  NonTrivialScanFunctor(NonTrivialScanFunctor const &) = default;
+  NonTrivialScanFunctor(NonTrivialScanFunctor &&)      = default;
+  NonTrivialScanFunctor &operator=(NonTrivialScanFunctor &&) = default;
+  NonTrivialScanFunctor &operator=(NonTrivialScanFunctor const &) = default;
+  ~NonTrivialScanFunctor() {}
+};
```

Comment:
The name does not clearly indicate that this is not only about not being trivial but also about no having the host device annotations on the destructor.  Plz comment or come up with a better name